oracle/controllers/backupcontroller/backup_controller.go

```diff
@@ -463,3 +444,33 @@ var preflightCheck = func(ctx context.Context, r *BackupReconciler, namespace, i
 	r.Log.Info("preflight check: physical backup, external LB service is ready", "succeededExecCmd#:", 1, "svc", svc.Name)
 	return nil
 }
+
+// checkLROStatus checks the status of a long-running operation (LRO) backup. LRO's are done through the db agent and thus needs to be monitored periodically.
+func (r *BackupReconciler) checkLROStatus(ctx context.Context, req ctrl.Request, backup *v1alpha1.Backup, rc *v1.Condition, inst *v1alpha1.Instance, log logr.Logger) (ctrl.Result, error, bool) {
+	id := lroOperationID(backup)
+	operation, err := controllers.GetLROOperation(r.ClientFactory, ctx, r, req.Namespace, id, backup.Spec.Instance)
+	if err != nil {
+		log.Error(err, "GetLROOperation error")
+		return ctrl.Result{}, err, true
+	}
+	if operation.Done {
+		log.Info("LRO is DONE", "id", id)
+		if operation.GetError() != nil {
+			log.Error(fmt.Errorf(operation.GetError().GetMessage()), "backup failed")
+			r.Recorder.Event(backup, corev1.EventTypeWarning, "BackupFailed", operation.GetError().GetMessage())
+			backup.Status.Conditions = k8s.Upsert(backup.Status.Conditions, k8s.Ready, v1.ConditionFalse, k8s.BackupFailed, operation.GetError().GetMessage())
+		} else {
+			r.Recorder.Eventf(backup, corev1.EventTypeNormal, "BackupCompleted", "BackupId:%v, Elapsed time: %v", backup.Status.BackupID, k8s.ElapsedTimeFromLastTransitionTime(rc, time.Second))
+			backup.Status.Conditions = k8s.Upsert(backup.Status.Conditions, k8s.Ready, v1.ConditionTrue, k8s.BackupReady, "")
+		}
+		if err := r.updateBackupStatus(ctx, backup, inst); err != nil {
+			log.Error(err, "failed to update the backup resource", "backup", backup)
+			return ctrl.Result{}, err, true
+		}
+		_ = controllers.DeleteLROOperation(r.ClientFactory, ctx, r, req.Namespace, id, backup.Spec.Instance)
```

Comment:
nit: we can have `return ctrl.Result{}, nil, false` here and get rid of the else below. I.e., 

if operation.Done {
 ...
  return ctrl.Result{}, nil, false
}

log.Info("LRO is in progress", "id", id)
return ctrl.Result{RequeueAfter: time.Minute}, nil, true